src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-list/smb-cluster-list.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+
+import _ from 'lodash';
+import { Subscription } from 'rxjs';
+
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { TableComponent } from '~/app/shared/datatable/table/table.component';
+import { Icons } from '~/app/shared/enum/icons.enum';
+import { CdTableAction } from '~/app/shared/models/cd-table-action';
+import { CdTableColumn } from '~/app/shared/models/cd-table-column';
+import { CdTableSelection } from '~/app/shared/models/cd-table-selection';
+import { Permission } from '~/app/shared/models/permissions';
+import { ListWithDetails } from '~/app/shared/classes/list-with-details.class';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { SmbService } from '~/app/shared/api/smb.service';
+import { CdTableFetchDataContext } from '~/app/shared/models/cd-table-fetch-data-context';
+
+@Component({
+  selector: 'cd-smb-cluster-list',
+  templateUrl: './smb-cluster-list.component.html',
+  styleUrls: ['./smb-cluster-list.component.scss']
+})
+export class SmbClusterListComponent extends ListWithDetails implements OnInit {
+  @ViewChild('table', { static: true })
+  table: TableComponent;
+
+  columns: CdTableColumn[];
+  permission: Permission;
+  selection = new CdTableSelection();
+  summaryDataSubscription: Subscription;
+  viewCacheStatus: any;
+  smbClusters: any[];
```

Comment:
Lets start with an interface for the `SMBCluster`. Preferably define it with the items that are expected to be shown in UI. So far I see
```
name, authMode, intent
```

and maybe
```
placement (like we show in the services if its useful)?
```

what do you think @Pegonzal ? i don't have any preferences here..
```suggestion
  smbClusters: SMBCluster[];
```

Reply:
Yep, we should look to implement SMB interfaces. For the SMBCluster I'd say:

```
cluster_id, auth_mode, intent, domain_settings: { realm, join_sources_ref }, user_group_settings, custom_dns and placement
```

`domain_settings` and `user_group_settings` should be optional as are dependant on the `auth_mode`